kubectl-fdb/cmd/profile_analyzer.go

```diff
@@ -0,0 +1,150 @@
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"log"
+	"strconv"
+	"text/template"
+
+	"github.com/spf13/cobra"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+)
+
+type profileConfig struct {
+	Namespace   string
+	ClusterName string
+	JobName     string
+	CommandArgs string
+}
+
+func newProfileAnalyzerCmd(streams genericclioptions.IOStreams) *cobra.Command {
+	o := newFDBOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:   "analyze-profile",
+		Short: "Analyze FDB shards to find the busiest team",
+		Long:  "Analyze FDB shards to find the busiest team",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			kubeClient, err := getKubeClient(o)
+			if err != nil {
+				return err
+			}
+
+			clusterName, err := cmd.Flags().GetString("fdb-cluster")
+			if err != nil {
+				return err
+			}
+			startTime, err := cmd.Flags().GetString("start-time")
+			if err != nil {
+				return err
+			}
+			endTime, err := cmd.Flags().GetString("end-time")
+			if err != nil {
+				return err
+			}
+			topRequests, err := cmd.Flags().GetInt("top-requests")
+			if err != nil {
+				return err
+			}
+			templateName, err := cmd.Flags().GetString("template-name")
+			if err != nil {
+				return err
+			}
+
+			namespace, err := getNamespace(*o.configFlags.Namespace)
+			if err != nil {
+				return err
+			}
+
+			return runProfileAnalyzer(kubeClient, namespace, clusterName, startTime, endTime, topRequests, templateName)
+		},
+		Example: `
+# Run the profiler for cluster-1. We require --cluster option explicitly because analyze commands take lot many arguments.
+kubectl fdb analyze-profile -c cluster-1 --start-time "01:01 20/07/2022 BST" --end-time "01:30 20/07/2022 BST" --top-requests 100
+`,
+	}
+	cmd.SetOut(o.Out)
+	cmd.SetErr(o.ErrOut)
+	cmd.SetIn(o.In)
+
+	cmd.Flags().StringP("fdb-cluster", "c", "", "cluster name for running hot shard tool.")
+	cmd.Flags().String("start-time", "", "start time for the analyzing transaction '01:30 30/07/2022 BST'")
+	cmd.Flags().String("end-time", "", "end time for analyzing the transaction '02:30 30/07/2022 BST'")
+	cmd.Flags().String("template-name", "", "Name of the Job template")
+	cmd.Flags().Int("top-requests", 100, "")
+	err := cmd.MarkFlagRequired("fdb-cluster")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = cmd.MarkFlagRequired("start-time")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = cmd.MarkFlagRequired("end-time")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = cmd.MarkFlagRequired("template-name")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	o.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func runProfileAnalyzer(kubeClient client.Client, namespace string, clusterName string, startTime string, endTime string, topRequests int, templateName string) error {
+	pc := profileConfig{
```

Comment:
```suggestion
	config := profileConfig{
```

I think `pc` it not an expressive variable name in this context.